compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -555,14 +574,21 @@ fn invalid_placeholder_type_error(
 fn report_missing_placeholders(
     ecx: &mut ExtCtxt<'_>,
     unused: Vec<(Span, &str)>,
+    duplicate_explicit_arg: Vec<Span>,
     detect_foreign_fmt: bool,
     str_style: Option<usize>,
     fmt_str: &str,
     fmt_span: Span,
 ) {
+    let mut dup_exist = false;
     let mut diag = if let &[(span, msg)] = &unused[..] {
         let mut diag = ecx.struct_span_err(span, msg);
         diag.span_label(span, msg);
+        for sp in &duplicate_explicit_arg {
+            if !dup_exist && span == *sp {
```

Comment:
I would like to rename these, because it is not about duplicates (`format!("{x} {x}")`), but rather about *redundant* args (`format!("{x}", x)`).